setup.md

```diff
@@ -269,8 +269,124 @@ TODO: Explain one of the following:
 * What significance does this have to the user?
 * What's in it for the user? WIIFM (What's in it for me? (for the user))
 
-### To add readline support to Fennel
+The command-line REPL that comes with the `fennel` works out of the box, but
+the built-in line-reader is very limited in user experience. Adding
+[GNU Readline](https://tiswww.case.edu/php/chet/readline/rltop.html) support
+enables such user-friendly features as
+- Tab completion on the REPL that can complete on all locals, macros, and special forms
+- A rolling history buffer, which can be navigated, searched (`ctrl+r`), and optionally
+persisted to disk so you can search input from previous REPL sessions.
+- emacs (default) or vi emulation via readline's custom support for better line
+navigation
+- Optional use of additional readline features in `~/.inputrc`, such as blinking
+on matched parentheses or color color output
 
-1. TODO
-2. TODO
-3. TODO
+### Requirements for readline support
+
+All you need to enable readline support is:
+- GNU Readline installed on your system (you may already have it!)
+- [readline.lua](https://pjb.com.au/comp/lua/readline.html) lua bindings to libreadline
+
+The Fennel CLI REPL will automatically load and use the readline bindings when
```

Comment:
Let's call it "the stock Fennel REPL" or "the `fennel` command's REPL" here instead of "The Fennel CLI REPL".

Reply:
Oh man that's so much better. I keep saying Fennel CLI REPL to keep it distinct from an nrepl or editor repl, but it feels awkward every time!